### ISSUE TITLE
Nans?

### ISSUE THREAD

**Issue by parkso** [ivasiljevic/fgan#1]
Hi , I have few questions.

I just downloaded and ran your code. 
I get an error after some ~= 500 epochs.
Is this because it is running for too long?

> Epoch: 498
> [[ 78.81487274]
>  [ 78.81487274]
>  [ 78.81487274]
>  ...,
>  [ 78.81487274]
>  [ 78.81487274]
>  [ 78.81487274]]
> ('Percent wrong by D: ', 0.0)
> ('Current sigma = ', array([ 0.00022991], dtype=float32))
> ('Current mu    = ', 1.3912925)
> 
> Epoch: 598
> [[ nan]
>  [ nan]
>  [ nan]
>  ...,
>  [ nan]
>  [ nan]
>  [ nan]]
> fgan.py:98: RuntimeWarning: invalid value encountered in less_equal
>   print("Percent wrong by D: ",np.mean(d_pred<=0.5))
> ('Percent wrong by D: ', 0.0)
> ('Current sigma = ', array([ nan], dtype=float32))
> ('Current mu    = ', nan)

Thanks

**Comment by ivasiljevic** [comment]
I switched to the total variation metric which seems to work fine, KL is still buggy.

**Comment by parkso** [comment]
This is for "KL" as function generator right?
So, then why do you add tf.log (1-  )  term to it??
Isn't it suppose to be -tf.reduce_sum((1-tf.zeros_like(D_)) * tf.exp(D_logits_-1)

**Comment by ivasiljevic** [comment]
You're right, for KL you'll have$$ f^{*}(g_f(v)) = \exp(v-1)$$.
